---
title: "RNAseq Data Processing & Analysing Course"
author: "Alex Gibbs"
#full-width: true
---

# Day 4 - Observing the outputs of the nf-core/differentialabundance pipeline and further analyses

---

## Recap from Day 3
---
---

- Yesterday we covered:
  - How to check the quality of the rnaseq data that we processed.
  - How to find the relevant outputs.
  - How exececute the differentialabundance pipeline.

---

<br>

---

## Outputs from the differentialabundance pipeline
---
---

- We should all have had an email from *SCW HAWK - HPC SERVICES* notifying you of a successful pipeline run.
- Sometimes the email function doesn't work. We can just log in and check ourselves.
- To check, we need to log back onto HAWK, load the tmux module, and then open the session that we created.
- NOTE: When using tmux, we need to make sure we are logged into the correct node. When we covered the log ins yesterday, we logged onto the cl1 node. The tmux session we made will only be present on the cl1 node. If we were to log in to the cl2 node, we wouldnt be able to find the tmux session.

<br>

```
#log onto HAWK
c.c1234567@hawklogin01.cf.ac.uk
PASSWORD

#move to the working directory (scratch)
cd /scratch/c.c1234567/rnaseq

#load tmux
module load tmux

#open our tmux session
tmux attach -t diff-abundance
```

<br>

- Upon opening the session, we should have a window that looks like this:

<img src="/assets/img/figure-19.png" alt="tmux completed run" width="1000"/>

- In this situation, the process took around 8 minutes. As all of us were using the pipeline at the same time, these times will likely vary. Download times will also depend on the size and number of files being downloaded.

**Report directory**
- This is where the main reporting output of the workflow is stored.
- Here we can find the report webpage (.html) and a zip file containing an R markdown file with all of the parameters to customise the reporting.

**Plots directory**
- Here we can find all of the stand-alone plots generated by the pipeline.
- Most of these plots are included in the webpage report.
- Plots include: 
    - QC: quality control plots from initial processing.
    - Exploratory: boxplots, PCA plots, correlation and dendograms. These are all plots that can be used to abundance distributions, sample clustering/correlations and observe any outliers.
    - Differential: volcano plots showing DEGs.

**Tables directory**
- This is where we can find the tables containing the count data:
    - Annotation: annotation matrix.
    - Processed Abundance: processed abundance values from initial processing. i.e. normalised counts from DESeq2.
    - Differential: Results of diferential analysis. filtered and unfiltered tables for expression.

**Pipeline Information directory**
- This directory contains all the files relevant to the pipeline that was run, such as a nextflow report, pipeline report.
- These files can be used to make a report if needed (handy when writing methods section).
- We donâ€™t need any of these files for the next stages.

**ShinyNGS App directory**
- This is where you will find the relevant files to run the Shiny app in R.
- We will cover how to run this below.

---
# So we've processed the samples and run the analysis pipeline... Now what?

- Now we can start exploring the data and begin answering our hypotheses.
- There are multiple things that we can do and multiple places we can start.
- I would highly reccommend that we start with the webpage report to have a general overview of the outputs.
- From there, we can then look at the Shiny app and use the built-in tools and filters to start exploring the data.
- We can then run additional analyses, such as Gene Set Enrichment Analysis (GSEA), Ingenuity Pathway Analysis (IPA), and further enrichment analysis tool on various web-tools.

## Webpage Report
- Lets start with the webpage report. We will firstly need to transfer the directory to our computers.
- We can do this with our file transfer apps, MobaXterm and FileZilla, or we could use code to transfer.
- Click this [link](INSERT LINK HERE!!!!!!*********) to see how to transfer using MobaXterm or FileZilla.
- See the code below for how to transfer using code:

```
Open a new shell window and cd to the directory where you want the directory to be transferred to.

#Move to downloads directory
cd /Users/Alex/Downloads

#transfer directory from HAWK to downloads
scp -r c.c1234567@hawklogin.cf.ac.uk:/scratch/c.c1234567/rnaseq/outputs/report .

#You will be asked to input your HAWK password and hit enter.
#Then the transfer will start.
```

- Now that we have the directory transferred, we can go ahead and navigate to it and open the report web page (study.html)


- The report contains all of the information about the differentialabundance analysis performed.
- At the top, we can see information about the Dataset, such as the sample name, strandedness and the condition columns that we added. Note that here I have a second condition column to bin the samples into cancer and control.
- Below this, we also have the contrasts table, indicating what contrasts we made. Note again here that I used the conditionTwo column to compare the Cancer samples to the Control samples.

<img src="/assets/img/figure-20.png" alt="Samples and contrasts tables" width="1000"/>

- Then we come onto the **Results** section. This is where all of the plots and data are presented.
- Firstly, we have the **counts** section which tells us how many input genes we had, and how many we had after filtering for low abundance.
- Then we have **exploratory analysis**, which shows us a range of plots that allow us to explore the data.
- **Box plots** shows us the distribution of abundance values across the samples. 
- Here we can see that we have 3 plots, one for each data type. We can see that normalisation and variance stabilisation has done a good gob. The purpose of this process is to adjust for differences in sequencing depth, control for different library sizes, minimize technical variability, and stabilise the variance across read counts.

<img src="/assets/img/figure-21.png" alt="boxplots" width="1000"/>

- **Density plots** give us an idea of overall expression levels across the dataset. Again, this section shows us the density plots across the 3 different data types.
- From the top 2 plots, it would be difficult to determine what is considered 'normal' expression here, as we have a lot of expression across the board. The point of normalisation and variance stabilisation is to make this easier to determine what is considered 'normal' expression and what is considered over expressed.
- We can see that in the variance stabilised plot, most of our genes are lowly/normally expressed (as indicated by the high peak on the left) and then we have a lwoer number of genes that would be considered over expressed.

<img src="/assets/img/figure-22.png" alt="density plots" width="1000"/>

- Next we have the **sample relationships** section. Here we will find PCA plots, Scree plots, association plots, dendrograms, and outlier detection plots.
- **PCA** plots show us how the samples cluster together. We can use this to see how similar or different samples are to eachother.
- The analysis is performed by using the top 500 most variable genes.
- We can see here that the Cancer samples (786-0) tightly cluster together on both PC1 and PC2 axes, whereas the Control samples (HK-2) are not so tightly clustered.
- However, there is a clear distinction between the two groups which is important when we are performing out differential analyses. We should see that there will be a higher number of differentially expressed genes in this analysis as theres a better separation between the groups.
- If we were to compare groups that didn't have a distinct separation, we would expect to see a lot less DEGs. We can also use this as an indicator of experimental conditions or techniques. 
  - If one sample in condition A is clustering with the samples in conditon B, was there a chance that somebody mislabelled the sample? 
  - If you drugged the same cell line for 1 hour and then compared drugged vs untreated, we would expect there to be very little difference between the groups and subsequently cluster close on a pca.
- When we look at the 3D PCA plot, where we are adding PC3 into the mix, we can see that there is more spread amongst the groups, but still a distinct separation between them.

<img src="/assets/img/figure-23.png" alt="PCA plots" width="1000"/>

- **Scree plots** show us the percent of variance that is explained by each principle component. We typically get most variance explained by PC1 and PC2, and then less and less explained by the latter PCs.
- We don't really need to pay any attention to this plot.


- **Association plot** shows the association between the principle components and the categorical covariates.
- I have not looked at this plot before, so can't comment much on it. I think this plot could potentially show more meaningful data if we had multiple conditions and we wanted to delve deeper into the PCA analyis.


- **Dendrogram plots** show us, similar to PCA, how the samples cluster in relation to eachother. 
- Like the PCA plots, we can see that each group cluster togeher and are clearly separated from each other.
- These plots can be handy to identify any potential outliers, as you will see them cluster apart from their respective groups.

<img src="/assets/img/figure-24.png" alt="Dendrograms" width="1000"/>

- **Outlier detection** plot is a handy plot that performs an analysis that can attempt to detect outliers in the dataset. 
- As we can see, all samples come up as FALSE, indicating that none are classed as outliers.

<img src="/assets/img/figure-25.png" alt="Outlier detection plot" width="1000"/>

-
